### PR TITLE
Update dependencies, allow Symfony 4

### PR DESCRIPTION
**Pull request by OndraM**
Hi, thanks for the tool - just trying it, however it was not compatible with Symfony 4 :).

I also updated all the outdated dependencies to latest versions :sparkles: .

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
-		"symfony/yaml": "~3.0"
+		"symfony/console": "~3.2|~4.0",
+		"symfony/yaml": "~3.0|~4.0"
 	},
 	"require-dev": {
-		"consistence/coding-standard": "2.0",
-		"jakub-onderka/php-parallel-lint": "0.9.2",
-		"phpstan/phpstan-shim": "0.8.4",
-		"phpunit/phpunit": "6.3.0"
+		"consistence/coding-standard": "^2.0",
```
Comment (mhujer) [inline composer.json]: What about locking dev dependencies to specific versions to prevent "random" build failures?

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
 	],
 	"require": {
 		"php": "~7.1",
-		"symfony/console": "~3.0|~4.0",
+		"symfony/console": "~3.2|~4.0",
```
Comment (mhujer) [inline composer.json]: Ups, it probably never worked with <3.2 😿

File: composer.json (hunk 1 above)
Comment (OndraM) [reply composer.json]: Well, at least you will fix it ASAP, no :)?

Its up to you, but I prefer not to worry about the upgrades, and just make sure it works (even regarding dev dependencies) with latest and lowest version by the constrains.

Comment (mhujer) [reply composer.json]: The downside is that the build may break when you need to fix something small (and unrelated) in the project in a few months.

File: composer.json [hunk 3]
```diff
@@ -20,14 +20,14 @@
...
 	},
 	"require-dev": {
-		"consistence/coding-standard": "2.0",
-		"jakub-onderka/php-parallel-lint": "0.9.2",
-		"phpstan/phpstan-shim": "0.8.4",
-		"phpunit/phpunit": "6.3.0"
+		"consistence/coding-standard": "^2.0",
+		"jakub-onderka/php-parallel-lint": "^1.0",
+		"phpstan/phpstan-shim": "0.9.2",
+		"phpunit/phpunit": "^7.0"
```
Comment (mhujer) [inline composer.json]: It should be also changed in https://github.com/mhujer/yaml-sort-checker/blob/0fcb205d8262f95552b650d105c00c5bf1dd78ba/phpunit.xml.dist#L5

Maybe it's worth switching to how the Doctrine does it - https://github.com/doctrine/doctrine2/blob/357005735833f709750cee59cf37dd5baf77c0a1/phpunit.xml.dist#L16 (`xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd`)

File: composer.json (hunk 1 above)
Comment (OndraM) [reply composer.json]: What we do is to set [Cron job](https://docs.travis-ci.com/user/cron-jobs/) which rebuild the master each day, so if something breaks, we know it soon and make sure it is fixed ASAP.

But it depends on your development process. I have no problem locking the versions if you say so :).

Comment (mhujer) [reply composer.json]: I think it is better to use the fixed versions here, as the project will probably be changed just rarely.

File: composer.json (hunk 3 above)
Comment (OndraM) [reply composer.json]: Nice trick, I did know about this :-)

**Comment by mhujer** [comment]
Thanks 👍

File: composer.json (hunk 1 above)
Comment (OndraM) [reply composer.json]: But the minor version should not change and break anything :).

However changed ;)

**Pull request merged**